pandas/core/groupby/ops.py

```diff
@@ -471,7 +471,12 @@ def _cython_operation(self, kind, values, how, axis, min_count=-1,
             if (values == iNaT).any():
                 values = ensure_float64(values)
             else:
-                values = values.astype('int64', copy=False)
+                try:
+                    values = values.astype('int64', copy=False, casting='safe')
```

Comment:
Hmm well this certainly works though I'm wondering if there's not a more comprehensive way that we should be handling (ex: adding the `casting="safe"` call to our `algos` templates)

cc @jreback and @jbrockmendel for any input

Reply:
I suppose this is something that occurs many places, as e.g. int and uint64 can not always be cast into int64 and thinking they can, just for satisfying is_integer_dtype is wrong. 

A few places around the code seems to have the same problem, e.g.:
pandas/core/algorithms.py:427
pandas/core/groupby/generic.py:1168

But as far as I can see, there isn't a general solution to this except for fixing the code in the respective places.

Reply:
can u take this code and make a function out of it and put where we defined ensure_float64
define

Reply:
@jreback I've created a function in dtypes/common.py now.